### DISCUSSION TITLE
Using a result source, tokens are not getting any values

### DISCUSSION THREAD

**Discussion by Jeroen00** [microsoft-search/pnp-modern-search#2711] [Q&A]
**Version used**
Ex: 4.8.0.1

**Describe the bug**
We are trying to use a Result Source in combination with the PnP Search Results web part. Configuring the source and using it in the web part is working. However, we need to use tokens in the result source to use a page property and the absolute URL. This is working when you use the tester in the result source but the PnP Search Results web part is then not returning results and the query modification result is not showing the values, only the tokens.

**To Reproduce**
Detailed steps to reproduce the behavior:
1. Create a result source and use a Page token in the query
2. Add the PnP Result Source web part and select the Result Source
3. Check the results and the query modification

**Expected behavior**
Tokens such as {Page.MyField1} and {Site.absoluteUrl} should be picked up by the PnP Search Results web part when used in a SharePoint Result Source.

**Result source configuration**
![image](https://user-images.githubusercontent.com/42589264/214308981-06701f11-db08-4289-b046-330cc88c2dc8.png)

**In SharePoint**
![image](https://user-images.githubusercontent.com/42589264/214309430-cbe45bf7-4bb1-4338-a535-04dd83995f64.png)

**Query modification**
![image](https://user-images.githubusercontent.com/42589264/214309476-c2733176-5878-4aad-a821-8cd0b9059d4d.png)

**Desktop**
 -Windows 10, Chrome browser

**Comment by wobba** [comment]
I don't think all query variables are supported in a result source, such as Page, when you do an API query from a modern page. As the page context is not passed. It only works on classic pages I believe. Put the query template in the web part instead.

**Reply by Jeroen00** [reply]
I do remember this worked in the past but then probably on classic pages (and it is also available in the query builder in the result source so a bit deceiving). Too bad because we are trying to have a reusable query that will be configured on hundreds of pages and for maintenance reasons we would like it to be centrally managed. Otherwise, updates will have to be done with PowerShell and that is much more complicated. 

Do you know of another way to achieve this?

**Reply by wobba** [reply]
When launching query builder from a classic page, then it pulls that data. For modern pages I don't see any other approach today, as the values has to be part of the query. In theory we could pass in all page props on the query so you can use them as query variables in a result source - but that means a bigger payload. I'll give it some thought and could be a web part option. **"Include page metadata for use as query variable in a result source"**.

Found my old post explaining what works OOB in REST https://www.techmikael.com/2015/07/sharepoint-rest-do-support-query.html but you can inject data as well to be picked up. Have to dig up some old stuff to figure out how :D

**Reply by wobba** [reply]
Here it is... testing via the query tool - so we can do something similar.
<img width="1194" alt="image" src="https://user-images.githubusercontent.com/51104/214790950-35de61ac-08c3-482b-9845-5aeb236555df.png">
